python/IECoreDelightTest/RendererTest.py

```diff
@@ -391,6 +391,29 @@ def testSphere( self ) :
 		self.__assertInNSI( '"width" "v float" 1 4', nsi )
 		self.__assertNotInNSI( '"N"', nsi )
 
+	def testProcedural( self ) :
+
+		r = GafferScene.Private.IECoreScenePreview.Renderer.create(
+			"3Delight",
+			GafferScene.Private.IECoreScenePreview.Renderer.RenderType.SceneDescription,
+			str( self.temporaryDirectory() / "test.nsia" ),
+		)
+
+		r.object(
+			"testProcedural",
+			IECoreScene.ExternalProcedural(),
+			r.attributes( IECore.CompoundObject() ),
+		)
+
+		r.render()
+		del r
+
+		nsi = self.__parse( self.temporaryDirectory() / "test.nsia" )
```

Comment:
This `self.__parse()` thing dates back all the way to when I first made GafferDelight, and really isn't very good, since it leaves us just dealing with text. Eric has since added `self.__parseDict()` which is much smarter and returns a dictionary-based representation of all the nodes, which we can use to make much  more robust assertions. I think it'd be worth using that here, and beefing up the assertions to cover filename, type, and additional custom parameters.

Reply:
Switched the test to use `self.__parseDict()`, `self.assertIn()` and `self.assertEqual()` testing the NSI procedural node `filename`, `type` and custom parameters.